apps/ex_nvr/lib/ex_nvr/recordings.ex

```diff
@@ -76,6 +76,90 @@ defmodule ExNVR.Recordings do
     )
   end
 
+  @spec delete_oldest_recordings(Device.t(), integer()) ::
+          {:ok, Ecto.Schema.t()} | {:error, Ecto.Changeset.t()}
+  def delete_oldest_recordings(device, limit) do
+    Multi.new()
+    # fetch recordings
+    |> Multi.run(:recordings, fn repo, _params ->
```

Comment:
Add an import at the start of the module `import Ecto.Query` and update this function to the following:
```elixir
@spec delete_oldest_recordings(Device.t(), integer()) :: :ok | {:error, Ecto.Changeset.t()}
  def delete_oldest_recordings(device, limit) do
    recordings = Recording.oldest_recordings(device.id, limit) |> Repo.all()

    oldest_recording_query = Recording.oldest_recordings(device.id, 1)
    delete_recordings_query = from(r in Recording, where: r.id in ^Enum.map(recordings, & &1.id))

    base_run_query = from(r in Run, where: r.device_id == ^device.id)
    oldest_run_query = from(r in base_run_query, order_by: r.start_date, limit: 1)

    Multi.new()
    |> Multi.delete_all(:recordings, delete_recordings_query)
    |> Multi.one(:oldest_recording, oldest_recording_query)
    |> Multi.delete_all(:runs, fn %{oldest_recording: recording} ->
      from(r in base_run_query, where: r.end_date < ^recording.start_date)
    end)
    |> Multi.one(:oldest_run, oldest_run_query)
    |> Multi.update(:run, fn %{oldest_run: run, oldest_recording: recording} ->
      Run.changeset(run, %{start_date: recording.start_date})
    end)
    |> Repo.transaction()
    |> case do
      {:ok, _changes} ->
        :telemetry.execute([:ex_nvr, :recording, :delete], %{count: length(recordings)}, %{
          device_id: device.id
        })

      {:error, _, changeset, _} ->
        {:error, changeset}
    end
end
```

I haven't tested it, you should add tests for this in `recordings_test.exs`.
Add also a changeset to `Run`